pkg/chainreader/contract_reader.go

```diff
@@ -0,0 +1,111 @@
+package chainreader
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query/primitives"
+)
+
+// ContractReaderByIDs wraps types.ContractReader to allow the caller to set custom contractIDs with Bind.
+type ContractReaderByIDs interface {
+	// Bind accepts a map of bound contracts where map keys are custom contractIDs to be used for calling the interface.
+	Bind(ctx context.Context, bindings map[string]types.BoundContract) error
+	Unbind(ctx context.Context, bindings map[string]types.BoundContract) error
+	GetLatestValue(ctx context.Context, contractID, readName string, confidenceLevel primitives.ConfidenceLevel, params, returnVal any) error
+	BatchGetLatestValues(ctx context.Context, request BatchGetLatestValuesRequestByCustomID) (BatchGetLatestValuesResultByCustomID, error)
+	QueryKey(ctx context.Context, contractID string, filter query.KeyFilter, limitAndSort query.LimitAndSort, sequenceDataType any) ([]types.Sequence, error)
+}
```

Comment:
Rather than create an interface, I think you can make `contractReaderByIDs` public